### ISSUE TITLE
Connecting to webdav with snap install?

### ISSUE THREAD

**Issue by LDWme** [nextcloud/nextcloud-snap#514]
Hello,
I have just followed the digital ocean tutorial on how to install nextcloud with snap on a clean Ubuntu server.
I have installed it just fine and seems to all be working great, my only issue is I can’t seem to webdav from Windows creating a new network share to my server. The servers running locally and I’m using the internal IP to connect. I thought it was an issue with windows at first but I started a next cloud trial from the home page and connected to that with WebDAV just fine, I am now thinking maybe because I install it with snap I need a different path when adding my share?

Thanks.

TLDR: Whats the Webdav URL for snap installs of NextCloud?

**Comment by QuaySo** [comment]
On Windows I use this:
`net use z: https://example.com/remote.php/dav /user:<USERNAME> <PASSWORD>`

**Comment by LDWme** [comment]
When I do the windows command I get:
"System error 1244 has occurred.

The operation being requested was not performed because the user has not been authenticated."

I have tried adding as a network drive with many different paths such as:
davs://example.com/remote.php/dav/files/USERNAME/
https://example.com/remote.php/dav/files/<USERNAME>

Nothing seems to be working.

**Comment by kyrofa** [comment]
Any chance you have two factor authentication enabled?

**Comment by LDWme** [comment]
No, it's a fresh install all settings are still at default.
It's really got me stumped.

**Comment by kyrofa** [comment]
@LDWme I'm really sorry, I simply don't have a Windows box with which to test. Perhaps @QuaySo can be of more assistance.

**Comment by LDWme** [comment]
@kyrofa Well thanks for attempting haha, im still trying to look my end. Think maybe it's something to do with my firewall? I run pfsense maybe somethings stopping it there?
I made a trial next cloud on their website and can webdav to that just fine.

**Comment by kyrofa** [comment]
I doubt it has anything to do with the firewall if you're able to do it with the trial (assuming the snap is using standard ports, is it?). Make sure you're not including the `/nextcloud` folder name as part of the URL, the snap doesn't use that, but I seem to remember the trial does.

Have you read [this](https://forums.iis.net/t/1176106.aspx?WebDAV+over+SSL)? Do you have HTTPS setup on the snap? If so, how? Can you verify the passive FTP thing?

**Comment by LDWme** [comment]
I set up my server the same as spoke about in this DO [guide](https://www.digitalocean.com/community/tutorials/how-to-install-and-configure-nextcloud-on-ubuntu-16-04).
Yes, it uses default ports and it does use self-signed SSL which the setup is also in the guide above.

**Comment by kyrofa** [comment]
Ah, I bet it's the fact that it's self-signed is throwing things off, in that article someone mentioned that using an invalid cert resulted in this error. Any chance you can use Let's Encrypt instead? You can also try disabling HTTPS just to test, if you can change your password to something garbage and don't have anything overly important there.

**Comment by kyrofa** [comment]
There's [this article](https://moblog.wiredwings.com/archives/20110406/webdav-windows-7-and-self-signed-certificates-howto.html) as well.

**Comment by LDWme** [comment]
I actually did use let's encrypt to make it, think it's worth deploying a new server and reinstalling and just trying without an SSL? I have a local VM server so I could just do that to test?

**Comment by kyrofa** [comment]
Wait, I'm confused-- is it self-signed, or using Let's Encrypt?

**Comment by LDWme** [comment]
Ugh sorry its lets encrypt

**Comment by kyrofa** [comment]
I suspect that's your issue. Mind giving us your domain? If you're not comfortable giving that publicly (I wouldn't be), on my profile there's an email address. Email it to me and I'll take a look.

**Comment by QuaySo** [comment]
It works for me with LetsEncrypt and a app password generated in Nextcloud. I do use TOTP...
There must be something wrong with your certificate.

**Comment by LDWme** [comment]
@kyrofa Ill shoot it over in an email now. 
Thank you!

**Comment by kyrofa** [comment]
@LDWme it seems that the cert is indeed self-signed, so try this:

1. Disable HTTPS:
    $ sudo nextcloud.disable-https

2. Enable HTTPS using Let's Encrypt:

    $ sudo nextcloud.enable-https lets-encrypt

Does that succeed? Let me know once you've done that and I'll take another look.

**Comment by LDWme** [comment]
@kyrofa Thanks I have done this but still don't seem to be getting anything.

**Comment by kyrofa** [comment]
The cert looks good to me, it comes from Let's Encrypt now, and I get no warnings. Still no love from Windows, though?

**Comment by LDWme** [comment]
Nope, what path do you recommend entering into windows for webdav?

**Comment by QuaySo** [comment]
@LDWme Do you use the `net use` command, or are you trying to connect the webdav share using the Explorer interface?

**Comment by LDWme** [comment]
@QuaySo Iv tried both

**Comment by QuaySo** [comment]
what is the URL you're using? `davs://` will not work on windows. Can you share the full URL?

**Comment by LDWme** [comment]
@QuaySo I have mainly been trying the one nextcloud gives me:
https://example.com/remote.php/webdav/
and:
https://example.com/nextcloud/remote.php/dav/files/USERNAME/

**Comment by QuaySo** [comment]
You should use the second, and leave out the `nextcloud` part. USERNAME should be substituted with lewis, or whatever username you use.

**Comment by LDWme** [comment]
Yea, sorry that's what Iv been using.

**Comment by LDWme** [comment]
@QuaySo I have no idea what changed but I just tried it again and it seems to connect! But it's not displaying any files from the server or letting me upload files, it's giving me an unexpected error when moving the file. This is a good start!

**Comment by fpernice518** [comment]
Any updates on this (or similar issues)? I'm having the same issue, but I'm definitely using a self-signed certificate.

**Comment by Koltsz** [comment]
> Any chance you have two factor authentication enabled? That has bitten me in the past with similar error messages, I needed to use a token ("app password") instead.

Thank you for this, I know its old but damn it was driving me nuts!. thank you again

**Comment by jumoog** [comment]
https://example.com/remote.php/webdav/ didn't worked for me https://example.com/remote.php/dav/files/USERNAME/ wored!
